plugins/modules/boot.py

```diff
@@ -384,15 +384,22 @@ def main():
                 option = module.params[option_name][option_key]
                 if option is None or option == []:
                     continue
-                data[data_key] = option
+
+                if isinstance(option, list):
+                    data[data_key+'[]'] = option
+                else:
+                    data[data_key] = option
             if existing.get('active'):
                 # Idempotence check
                 needs_change = False
                 for option_key, (result_key, data_key) in options.items():
                     should = module.params[option_name][option_key]
                     if should is None:
                         continue
-                    has = existing.get(data_key)
+                    if option_key == 'authorized_keys':
+                        has = list(map(lambda x: x['key']['fingerprint'], existing.get(data_key)))
+                    else:
+                        has = existing.get(data_key)
```

Comment:
```suggestion
                    has = existing.get(data_key)
                    if option_key == 'authorized_keys':
                        has = [x['key']['fingerprint'] for x in has]
```
(It's too bad that the API docs do not provide any example or schema for how `authorized_key` looks like...)

Reply:
For the record: the list entries are of the form `{'key': {'fingerprint': 'aa:bb:cc:...', 'name': 'foo', 'size': 4096, 'type': 'RSA'}}`.

Reply:
(This also needs to be adjusted in the tests. They right now assume that this is a list of fingerprints.)